.circleci/config.yml

```diff
@@ -50,6 +50,35 @@ jobs:
           paths:
             - html-scipyorg
 
+      # Benchmarks -- run if:
+      #
+      # 1. On `master` branch
+      # 2. Any files under benchmarks/ have changed
+      # 3. The magic "[circle bencmark]" tag is in the commit message.
+      #
+      - run:
+          name: benchmarks
+          command: |
+            . venv/bin/activate
+            export SHELL=$(which bash)
+            if ! git remote -v | grep upstream ; then git remote add upstream git://github.com/scipy/scipy.git; fi
+            changed=$(git diff --name-only $CIRCLE_BRANCH $(git merge-base $CIRCLE_BRANCH upstream/master));
+            if [ "${CIRCLE_BRANCH}" == "master" ] || [[ `expr match "${changed}" ".*benchmarks/.*"` ]] || [[ $(git log -1 --pretty=%B) == *"[circle benchmark]"* ]]; then
+              arch=$(uname -m)
+              cpu=$(cat /proc/cpuinfo |grep -m 1 "model name"|cut -d' ' -f 3-)
+              ram=$(awk -F"[: ]+" '/MemTotal/ {print $2;exit}' /proc/meminfo)
+              os=$(uname -sr)
+              printf "{\n\"CircleCI\": {\n\"arch\": \"${arch}\",\n\"cpu\": \"${cpu}\",\n\"machine\": \"CircleCI\",\n\"os\": \"${os}\",\n\"ram\": \"${ram}\"\n},\n\"version\": 1\n}" | tee ~/.asv-machine.json
+              python -u runtests.py --bench [-q]
```

Comment:
> the numbers probably are quite meaningless, since it's run on some random CircleCI environment whose performance is not constant probably.

It's probably not constant, but I would be surprised if it differed by an order of magnitude from run to run. So it could at least flag potential egregious performance regressions. Eventually if this works we could make these persist for `CircleCI` then we could build up a history, too, which would be cool.

Do you have a recommended `asv` or `run.py` command? Maybe
```
./run.py run HEAD^^..HEAD
```
?

Though if the one-commit `runtests.py` took almost two hours, I'm guessing this will be too slow at almost 4 hours.